source/common/router/scoped_rds.h

```diff
@@ -191,6 +234,11 @@ class ScopedRdsConfigSubscription
   const envoy::config::core::v3::ConfigSource rds_config_source_;
   const std::string stat_prefix_;
   RouteConfigProviderManager& route_config_provider_manager_;
+
+  // RdsRouteConfigProvider by scope name.
+  absl::flat_hash_map<std::string, RdsRouteConfigProviderHelperPtr> route_provider_by_scope_;
+  // A map of (hash, scope-name), used to detect the key conflict between scopes.
```

Comment:
why reorg the code?

Reply:
because of the stats, when the helpers are destroyed, the call the stats in their dtor. so they must be destroyed first, otherwise we will have segment fault.